app/scenes/create_channel/create_channel.js

```diff
@@ -0,0 +1,355 @@
+
+import ErrorText from 'app/components/error_text';
+import FormattedText from 'app/components/formatted_text';
+import TextInputWithLocalizedPlaceholder from 'app/components/text_input_with_localized_placeholder';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import {Constants, RequestStatus} from 'service/constants';
+import EventEmitter from 'service/utils/event_emitter';
+
+import CreateChannelButton from './create_channel_button';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    const {height, width} = Dimensions.get('window');
+    return StyleSheet.create({
+        container: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg
+        },
+        scrollView: {
+            flex: 1,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.03),
+            paddingTop: 30,
+            height: height + (Platform.OS === 'android' ? 200 : 0)
+        },
+        errorContainer: {
+            position: 'absolute',
+            width
+        },
+        inputContainer: {
+            marginTop: 10,
+            borderTopWidth: 1,
+            borderBottomWidth: 1,
+            borderTopColor: changeOpacity(theme.newMessageSeparator, 0.5),
+            borderBottomColor: changeOpacity(theme.newMessageSeparator, 0.5),
+            backgroundColor: theme.centerChannelBg
+        },
+        input: {
+            color: theme.centerChannelColor,
+            fontSize: 14,
+            height: 40,
+            paddingHorizontal: 15
+        },
+        title: {
+            fontSize: 14,
+            color: theme.centerChannelColor,
+            marginLeft: 15
+        },
+        optional: {
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+            fontSize: 14,
+            marginLeft: 5
+        },
+        helpText: {
+            fontSize: 14,
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+            marginTop: 10,
+            marginHorizontal: 15
+        }
+    });
+});
+
+class CreateChannel extends PureComponent {
+    static propTypes = {
+        intl: intlShape.isRequired,
+        createChannelRequest: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        channelType: PropTypes.string,
+        subscribeToHeaderEvent: React.PropTypes.func.isRequired,
+        unsubscribeFromHeaderEvent: React.PropTypes.func.isRequired,
+        actions: PropTypes.shape({
+            goBack: PropTypes.func.isRequired,
+            handleCreateChannel: PropTypes.func.isRequired
+        })
+    };
+
+    static defaultProps = {
+        channelType: Constants.OPEN_CHANNEL
+    };
+
+    static navigationProps = {
+        renderLeftComponent: (props, emitter, theme) => {
+            return (
+                <TouchableOpacity
+                    style={{flex: 1, paddingHorizontal: 15, justifyContent: 'center'}}
```

Comment:
You can probably still use getStyleSheet here since the `theme` provided here should be the same